### DISCUSSION TITLE
⛔ Transaction Error  Error: transaction underpriced

### DISCUSSION THREAD

**Discussion by jtruji68** [lmusarella/Pancakeswap-Prediction-Bot-2023#28] [Q&A] [answered]
Hi, I tried to run the bot in real mode, but apparently the default fees are not enough because it throws:

`⛔ Transaction Error [⌚ 16:41][215180][-32000} ] => Error: transaction underpriced
    at getResult (/home/juan/cr/Pancakeswap-Prediction-Bot-2023/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:191:21)
    at processJsonFunc (/home/juan/cr/Pancakeswap-Prediction-Bot-2023/node_modules/@ethersproject/web/lib/index.js:356:22)
    at /home/juan/cr/Pancakeswap-Prediction-Bot-2023/node_modules/@ethersproject/web/lib/index.js:288:46
    at step (/home/juan/cr/Pancakeswap-Prediction-Bot-2023/node_modules/@ethersproject/web/lib/index.js:33:23)
    at Object.next (/home/juan/cr/Pancakeswap-Prediction-Bot-2023/node_modules/@ethersproject/web/lib/index.js:14:53)
    at fulfilled (/home/juan/cr/Pancakeswap-Prediction-Bot-2023/node_modules/@ethersproject/web/lib/index.js:5:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: -32000,
  data: undefined
}`

When I try to  make a $1 bet. But i cant find the estimate gas variable on the configuration file.
Thanks

**Comment by lmusarella** [comment]
Hi, thanks for the report, you're not the first. I'll take care of it as
soon as I can

Il dom 15 ott 2023, 23:52 David Trujillo ***@***.***> ha
scritto:

**Reply by jtruji68** [reply]
@lmusarella  Thanks, from what i've seen its not about the gas values themselves, because its using the default one (0x59682f00 or 1500000000). And according to this post https://github.com/ethers-io/ethers.js/discussions/3018 , it has to be an incorrect nonce the one that is generated with getTransactionCount("pending") or it may be due to the maxPriorityFeePerGas not high enough. But I wonder if you have the same problem, Im using a trustwallet wallet, and a quicknode endpoint. So I think this is a good place to start. what do you think?

**Reply by jtruji68** [reply]
wait, i just checked the values and the Gwei is at 3 in pancakeswap, when the bot is using just 1.5. Can you tell me where I can harcode the values so I can check faster?

**Reply by jtruji68** [reply, marked as answer]
Nevermind, I realize is the node provider, some quicknode endpoints are weird with their gas fees, sometimes even lower than the current market, the solution was to change provider URL, i got one from getblock .com and it works now

**Comment by jtruji68** [comment]
> Nevermind, I realize is the node provider, some quicknode endpoints are weird with their gas fees, sometimes even lower than the current market, the solution was to change provider URL, i got one from getblock .com and it works now

Update: I dont use getblock.com anymore, i switch between `"https://bscrpc.com"` and `"https://bsc-mainnet-rpc.allthatnode.com"` when one is getting too many errors.
